example/javascriptlib/basic/1-simple/build.mill

```diff
@@ -1,25 +1,125 @@
+
+  def sourceForTask = Task.Anon {
+    val destination = T.dest.toString()
+    val destSource = destination match {
+      case x if !x.contains("test") => os.walk(sources().path).filter(_.ext == "ts").map(PathRef(_))
+      case _ => (os.walk(sources().path) ++ os.walk(testPath().path) ++ IndexedSeq(
+          testConfigPath
+        )).filter(_.ext == "ts").map(PathRef(_))
+    }
+
+    destSource
+  }
+
+  def tsConfig = Task.Anon {
+    val nodeTypes = npmInstall().path / "node_modules/@types"
+    val javascriptOut = Task.dest / "javascript"
+    val declarationsOut = Task.dest / "declarations"
+
+    val upstreamPaths =
+      for (((jsDir, dTsDir), mod) <- Task.traverse(moduleDeps)(_.compile)().zip(moduleDeps))
+        yield (mod.millSourcePath.subRelativeTo(build.millSourcePath) + "/*", dTsDir.path)
+
+    val allPaths = upstreamPaths ++ Seq("*" -> sources().path, "*" -> npmInstall().path)
+    os.write(
+      Task.dest / "tsconfig.json",
+      ujson.Obj(
+        "compilerOptions" -> ujson.Obj(
+          "outDir" -> javascriptOut.toString,
+          "declaration" -> true,
+          "declarationDir" -> declarationsOut.toString,
+          "typeRoots" -> ujson.Arr(nodeTypes.toString),
+          "paths" -> ujson.Obj.from(allPaths.map { case (k, v) => (k, ujson.Arr(s"$v/*")) })
+        ),
+        "files" -> sourceForTask().map(_.path.toString)
+      )
+    )
+  }
+
+  override def compile: T[(PathRef, PathRef)] = Task {
+    tsConfig()
+    os.call(npmInstall().path / "node_modules/typescript/bin/tsc")
+    val javascriptOut = T.dest / "javascript"
+    val declarationsOut = T.dest / "declarations"
+    (PathRef(javascriptOut), PathRef(declarationsOut))
+  }
+
+  def typeCheck: T[Unit] = Task {
+    tsConfig()
+    os.call((
+      npmInstall().path / "node_modules/typescript/bin/tsc",
+      "--noemit"
+    ))
+  }
+}
+
+object foo extends SimpleMod {
+  object bar extends SimpleMod {
     def npmDeps = Seq("immutable@4.3.7")
   }
 }
 
-object qux extends TypeScriptModule {
+object qux extends SimpleMod {
   def moduleDeps = Seq(foo, foo.bar)
+
+  def test = Task {
+    tsConfig()
+    os.call(npmInstall().path / "node_modules/typescript/bin/tsc")
+
+    val javascriptOut = Task.dest / "javascript"
+
+    // env
+    // note: ' npmInstall().path / "node_modules" ' required in NODE_PATH for jest to find preset: ts-jest
+    val env = Map("NODE_PATH" -> Seq(
+      ".",
+      javascriptOut,
+      npmInstall().path,
+      npmInstall().path / "node_modules"
+    ).mkString(":"))
+
+    os.call(
+      ("node", npmInstall().path / "node_modules/jest/bin/jest.js", javascriptOut.toString()),
+      stdout = os.Inherit,
+      env = env
+    )
+  }
 }
 
-// Documentation for mill.javascriptlib
+// Documentation for mill.example.javascriptlib.basic
 
 /** Usage
 
-> mill qux.run James Bond prof
+> mill dist.run example/javascriptlib/basic/1-simple qux.run James Bond prof
```

Comment:
@monyedavid in the example tests you can use `mill` without `dist.run`. Please look at the other example tests for reference, the readme.adoc for how to run these examples, and make sure they are green

Reply:
got it